category_encoders/target_encoder.py

```diff
@@ -104,22 +105,33 @@ def __init__(self, verbose=0, cols=None, drop_invariant=False, return_df=True, h
         self.mapping = None
         self._mean = None
         if hierarchy:
-            if self._depth(hierarchy) > 2:
-                raise ValueError('Hierarchy mapping contains too many levels')
-        self.hierarchy = hierarchy
+            self.hierarchy = {}
+            self.hierarchy_depth = {}
+            for switch in hierarchy:
+                D = fd.flatten(hierarchy[switch], inverse=True)
+                hierarchy_check = self._check_dict_key_tuples(D)
+                self.hierarchy_depth[switch] = hierarchy_check[1]
+                if not hierarchy_check[0]:
+                    raise ValueError('Hierarchy mapping contains different levels for key "' + switch + '"')
+                self.hierarchy[switch] = {(k if type(t) is tuple else t): v for t, v in D.items() for k in t}
```

Comment:
checking for types should be done using `isinstance(t, tuple)`

Reply:
Agreed.